File: C_Programming/Practise_Programs/PP_115.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_115
//
//  Description       : This program demonstrates modification of
//                      the first character of a string using a
//                      character pointer.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 14/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

//////////////////////////////////////////////////////////////////
//
//  Function Name     : strlenX()
//
//  Description       : It is used to modify the first character of
//                      the given string using a character pointer.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 14/08/2026
//
//////////////////////////////////////////////////////////////////

void strlenX(char *str)
{
    *str = 'A';
}

int main()
{
    char Arr[50] = {'\0'};
    int iRet = 0;

    printf("Enter string : \n");
    scanf("%[^'\n']s", Arr);  
    
    strlenX(Arr);

    printf("String is : %s\n", Arr);
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Ganesh
//
//  Sample Output :
//
//      Enter string :
//      String is : Aanesh
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_115.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_115
//
//  Description       : This program accepts a string from the user
//                      and passes it to a user-defined class method
//                      to display the received string.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 26/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class StringX
{
    //////////////////////////////////////////////////////////////////
    //
    //  Method Name      : Display()
    //
    //  Description      : It is used to display the string received
    //                      as a parameter.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 26/08/2026
    //
    //////////////////////////////////////////////////////////////////

    public void Display(String str)
    {
        System.out.println("Received String is : " + str);
    }
}

class PP_115
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        String str = null;

        StringX strobj = new StringX();

        System.out.println("Enter String : ");
        str = sobj.nextLine();

        strobj.Display(str);

        sobj.close();
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Ganesh
//
//  Sample Output :
//
//      Enter String :
//      Received String is : Ganesh
//
//////////////////////////////////////////////////////////////////